src/js/components/Diagram/StyledDiagram.js

```diff
@@ -1,12 +1,80 @@
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
 import { defaultProps } from '../../default-props';
+import { animationObjectStyle } from '../../utils/animations';
+
+const animationItemStyle = (animationType, theme) => {
+  if (typeof animationType === 'string') {
+    return animationObjectStyle({ type: animationType }, theme);
+  }
+  if (typeof animationType === 'object') {
+    return animationObjectStyle(animationType, theme);
+  }
+  if (typeof animationType === 'boolean') {
+    return animationObjectStyle({ type: 'draw' }, theme);
+  }
+  return '';
+};
+
+const availableAnimations = [true, 'draw', 'pulse'];
+
+const animationStyle = props => {
+  const animationType = props.animation.type || props.animation;
+
+  if (!availableAnimations.includes(animationType)) {
+    return 'Not available for Diagram';
+  }
+  if (animationType === 'draw' || animationType === true) {
+    return css`
+      path {
+        stroke-dasharray: 500;
+        stroke-dashoffset: 500;
+        animation: ${animationItemStyle(props.animation, props.theme)};
+      }
+    `;
+  }
+  return css`
+    animation: ${animationItemStyle(props.animation, props.theme)};
+  `;
+};
+
+const connectionStyle = (connections, theme) =>
+  connections.map((connection, index) => {
+    if (connection !== undefined && connection.props.animation !== undefined) {
+      const { type } = connection.props.animation;
+      const animationType = type || connection.props.animation;
+
+      if (!availableAnimations.includes(animationType)) {
+        return 'Not available for Diagram';
+      }
+      if (animationType === 'draw' || animationType === true) {
+        return css`
+          path:nth-child(${index + 1}) {
+            stroke-dasharray: 500;
+            stroke-dashoffset: 500;
+            animation: ${animationItemStyle(connection.props.animation, theme)};
+          }
+        `;
+      }
+      return css`
+        path:nth-child(${index + 1}) {
+          stroke-dasharray: 0;
+          stroke-dashoffset: 0;
+          animation: ${animationItemStyle(connection.props.animation, theme)};
+        }
+      `;
+    }
+    return '';
+  });
 
 const StyledDiagram = styled.svg`
   max-width: 100%;
   width: 100%;
   height: 100%;
 
+  ${props =>
```

Comment:
Nice!

Reply:
Maybe add a comment about why this needs to happen before the props.animation below? (otherwise connection level animations won't work correctly)

Reply:
Added the comment, after looking up what was recommended for commenting inside a styled component but general consensus seemed to just comment it like normal CSS. Let me know if it looks okay in the newest commit I pushed.